various 1.21.3 compatibility fixes

sylviameows opened the pull request:
running the client seems stable and no glaring issues are obvious to me anymore. 
however, i did notice that the `DestroyItemReset` config option wasn't saving and I wasn't able to fix this issue.

**[Changes]**
- migrated build phaser to use `/ptp [x] [y] [z] [pitch] [yaw]` (originally this was supposed to fix a disconnect when using the feature, but the issue was unrelated and has been fixed)

**[Fixes]**
- fixed all MiniMessage features to work for 1.21.3
- fixed Code Palette and Custom Chest GUI rendering for 1.21.3
- fixed chest highlighting feature 'blinking' in 1.21.3
- fixed an issue where `space` and `newline` tags were affecting the chat bar in negative ways.

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/Utility.java:
```diff
@@ -174,7 +174,7 @@ public static void sendMessage(Text message, @Nullable ChatType type) {
      */
     public static NbtString textToNBT(Text text) {
-        assert CodeClient.MC.world != null;
+        assert CodeClient.MC.world != null; // ??? do we need this
```
We do not

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/config/Config.java:
```diff
@@ -555,15 +555,15 @@ public YetAnotherConfigLib getLibConfig() {
                                 .controller(integerOption -> IntegerFieldControllerBuilder.create(integerOption).range(0, 100))
                                 .build())
-                        .option(Option.createBuilder(Config.DestroyItemReset.class)
+                        .option(Option.createBuilder(DestroyItemReset.class)
```
If you're looking to fix this, make sure the name of the option's variable on the top matches the name in that big list that serializes the options

sylviameows replied:
ah i see now, i was looking and looking and never noticed those names were different

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/dev/BuildPhaser.java:
```diff
@@ -151,6 +144,17 @@ private void finishClipping() {
+            var pitch = player.getPitch();
+            var yaw = player.getYaw();
+
+            player.networkHandler.sendChatCommand(String.format("ptp %s %s %s %s %s", x, y, z, pitch, yaw));
```
Remind me to send you the queue and ratelimiting thing for commands to keep codeclient as an intrusive as it has been

RedVortexDev reviewed (changes requested):
Thank you so much ❤️

sylviameows commented:
resolved the requested changes, here are the changes made:

**[Additions]**
- added rate limiting to codeclient sent commands (thanks red!)
- added `/ccclearqueue` to clear the command queue when it gets too long.

**[Changes]**
- enabled build phaser in build mode.

**[Fixes]**
- fixed DestroyItemMode config option not saving.

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/command/CommandSender.java:
```diff
@@ -0,0 +1,61 @@
+    */
+
+    public static void queue(String command) {
+        rateLimiter.increment();
```
You don't need to rate limit adding to the queue, it should only register an operation on the tick method where the commands are actually sent, if you'd like you could make an internal rate limiter but I don't think that's necessary

sylviameows replied:
ah gotcha, i was just copying what you sent me :P

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/Event.java:
```diff
@@ -52,10 +54,14 @@ public static <T extends PacketListener> void handlePacket(Packet<T> packet) {
 
     public static <T extends PacketListener> void onSendPacket(Packet<T> packet) {
         if (packet instanceof CommandExecutionC2SPacket command) {
+            CommandSender.registerCommandSend();
```
Just in case, are you able to confirm that command execution doesnt also trigger chat message packet (so we don't increment it twice)

sylviameows replied:
it does not increment twice

RedVortexDev reviewed (changes requested):
a bit about command queue

RedVortexDev commented on src/main/java/dev/dfonline/codeclient/dev/BuildPhaser.java:
```diff
@@ -104,14 +109,25 @@ public boolean onSendPacket(Packet<?> packet) {
         if (packet instanceof PlayerPositionLookS2CPacket move) {
 //            CodeClient.MC.getNetworkHandler().sendPacket(new TeleportConfirmC2SPacket(move.teleportId()));
-            return false;
+            CodeClient.LOGGER.info(move.change().deltaMovement().toString());
```
leftover debug :P

sylviameows replied:
oh oops lol

RedVortexDev reviewed (approved):
LGTM, merging to 1.21 branch.
